go/api/options/script_options.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package options
+
+import (
+	"sync"
+	"unsafe"
+)
+
+// #include "../../lib.h"
+import "C"
+
+// ScriptOptions represents options for script execution
+type ScriptOptions struct {
+	keys []string
+	args []string
+}
+
+// NewScriptOptions creates a new ScriptOptions with default values
+func NewScriptOptions() *ScriptOptions {
+	return &ScriptOptions{
+		keys: []string{},
+		args: []string{},
+	}
+}
+
+// WithKeys sets the keys for the script
+func (o *ScriptOptions) WithKeys(keys []string) *ScriptOptions {
+	o.keys = keys
+	return o
+}
+
+// WithArgs sets the arguments for the script
+func (o *ScriptOptions) WithArgs(args []string) *ScriptOptions {
+	o.args = args
+	return o
+}
+
+// GetKeys returns the keys for the script
+func (o *ScriptOptions) GetKeys() []string {
+	return o.keys
+}
+
+// GetArgs returns the arguments for the script
+func (o *ScriptOptions) GetArgs() []string {
+	return o.args
+}
+
+// Script represents a Lua script stored in Valkey/Redis
+type Script struct {
+	hash      string
+	isDropped bool
+	mu        sync.Mutex
+}
+
+// NewScript creates a new Script object
+func NewScript(code interface{}) *Script {
```

Comment:
```suggestion
func NewScript(code string) *Script {
```
and you can delete `stringOf` method

Reply:
Will this always be a string?

Reply:
Updated.